dds/idl/v8_generator.cpp

```diff
@@ -206,6 +428,37 @@ bool v8_generator::gen_struct(AST_Structure*, UTL_ScopedName* name,
       "    return OpenDDS::DCPS::copyToV8(*static_cast<const " << lname
                << "*>(source));\n"
       "  }\n\n"
+      "  void* fromV8(const v8::Local<v8::Object>& source) const\n"
+      "  {\n"
+      "    " << lname << "* result = new " << lname << "();\n"
+      "    OpenDDS::DCPS::copyFromV8(source, *result);\n"
+      "    return result;\n"
+      "  }\n\n"
+      "  void deleteFromV8Result(void* val) const\n"
+      "  {\n"
+      "    " << lname << "* delete_me = static_cast< " << lname << "*>(val);\n"
+      "    delete delete_me;\n"
+      "  }\n\n"
+      "  DDS::InstanceHandle_t register_instance_helper(DDS::DataWriter* dw, const void* data) const\n"
+      "  {\n"
+      "    " << lname << "DataWriter* dw_t = dynamic_cast<" << lname << "DataWriter*>(dw);\n"
```

Comment:
What when the dynamic cast fails?

Reply:
I'd hope the correct type would always get passed since this is generated code, but OK. It doesn't hurt to add a check and return nil / bad parameter.